File: solutions/go/MinimumSizeSubarraySum.go
import (
	"math"
)

func minSubArrayLen(target int, nums []int) int {
	var i, j, sum int

	length := len(nums)
	min := math.MaxInt32

	for j < length {
		sum += nums[j]
		j++

		for sum >= target {
			tmp := j - i
			if tmp < min {
				min = tmp
			}

			sum -= nums[i]
			i++
		}
	}

	if min == math.MaxInt32 {
		return 0
	}

	return min
}

File: solutions/c/MinimumSizeSubarraySum.c
int
minSubArrayLen(int target, int *nums, int len)
{
    int i, j, sum, min;

    min = INT_MAX;
    j = sum = 0;

    for (i = 0; i < len; ++i) {
        sum += nums[i];

        while (sum >= target) {
            if (i-j+1 < min) {
                min = i - j + 1;
            }

            sum -= nums[j++];
        }
    }

    return min == INT_MAX ? 0 : min;
}

